SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/SteamUnifiedMessages.cs

```diff
@@ -16,22 +14,31 @@ namespace SteamKit2
     /// </summary>
     public partial class SteamUnifiedMessages : ClientMsgHandler
     {
-        private readonly Dictionary<string, UnifiedService> _handlers = [ ];
+        private readonly ConcurrentDictionary<string, UnifiedService> _handlers = [ ];
 
         /// <summary>
         /// Creates a service that can be used to send messages and receive notifications via Steamworks unified messaging.
         /// </summary>
         /// <typeparam name="TService">The type of the service to create.</typeparam>
-        /// <returns>The instance to create requests.</returns>
+        /// <returns>The instance to create requests from.</returns>
         public TService CreateService<TService>() where TService : UnifiedService, new()
         {
             var service = new TService
             {
                 UnifiedMessages = this
             };
 
-            _handlers.Add( service.ServiceName, service);
-            return service;
+            return (_handlers.GetOrAdd( service.ServiceName, service ) as TService)!;
+        }
+
+        /// <summary>
+        /// Removes a service so it no longer can be used to send messages or receive notifications.
+        /// </summary>
+        /// <typeparam name="TService">The type of the service to remove.</typeparam>
+        public void RemoveService<TService>() where TService : UnifiedService, new()
+        {
+            var serviceName = new TService().ServiceName;
```

Comment:
Can ServiceName be made static? I don't know if `TService.ServiceName` is possible.